modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/client/ItThinClientCriteriaQueryTest.java

```diff
@@ -0,0 +1,96 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.runner.app.client;
+
+import static java.util.stream.Collectors.toList;
+import static org.apache.ignite.table.criteria.CriteriaQueryOptions.builder;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.nio.file.Path;
+import java.util.Map;
+import org.apache.ignite.internal.testframework.WorkDirectory;
+import org.apache.ignite.table.RecordView;
+import org.apache.ignite.table.Tuple;
+import org.hamcrest.FeatureMatcher;
+import org.hamcrest.Matcher;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+
+/**
+ * Thin client criteria query test.
+ */
+@SuppressWarnings("resource")
+public class ItThinClientCriteriaQueryTest extends ItAbstractThinClientTest {
+    /** {@inheritDoc} */
+    @Override
+    @BeforeAll
+    void beforeAll(TestInfo testInfo, @WorkDirectory Path workDir) throws InterruptedException {
+        super.beforeAll(testInfo, workDir);
+
+        populateData();
+    }
+
+    @Test
+    public void testBasicQueryCriteriaRecordBinaryView() {
+        var view = client().tables().table(TABLE_NAME).recordView();
+
+        var res = view.queryCriteria(null, null).stream().collect(toList());
+        assertThat(res, hasSize(3));
+
+        var ars = view.queryCriteriaAsync(null, null, builder().pageSize(2).build()).join();
+        assertEquals(2, ars.currentPageSize());
+    }
+
+    @Test
+    public void testBasicQueryCriteriaRecordPojoView() {
+        RecordView<TestPojo> view = client().tables().table(TABLE_NAME).recordView(TestPojo.class);
+
+        var res = view.queryCriteria(null, null).stream().collect(toList());
+        assertThat(res, hasSize(3));
+
+        var ars = view.queryCriteriaAsync(null, null, builder().pageSize(2).build()).join();
+        assertEquals(2, ars.currentPageSize());
+    }
+
+    private void populateData() {
+        var table = client().tables().table(TABLE_NAME).recordView();
+
+        table.insert(null, Tuple.create(Map.of(COLUMN_KEY, 1, COLUMN_VAL, "1")));
+        table.insert(null, Tuple.create(Map.of(COLUMN_KEY, 2, COLUMN_VAL, "2")));
+        table.insert(null, Tuple.create(Map.of(COLUMN_KEY, 3, COLUMN_VAL, "3")));
+    }
+
+    /**
+     * Creates a matcher for matching tuple value.
+     *
+     * @param valueMatcher Matcher for matching tuple value.
+     * @return Matcher for matching tuple value.
+     */
+    private static <T> Matcher<Tuple> tupleValue(String columnName, Matcher<T> valueMatcher) {
```

Comment:
is it required?

Reply:
Deleted. I will add this in another PR where it is used.